lib/msal-common/docs/multi-tenant-accounts.md

```diff
@@ -0,0 +1,184 @@
+
+> Warning: While MSAL can return a tenant-specific `AccountInfo` object for each tenant profile, tenant profiles are not actually different accounts, they are only representations of the same account in the different tenants the account has authentiated with. **If a user uses different accounts to authenticate to each tenant, these will not be linked as tenant profiles of eachother and will be treated as completely different accounts that cannot be used to access eachother's tokens**.
+
+## Usage
+
+### Authenticating with Multiple Tenants
+
+In order to authenticate with multiple tenants, you can use either `login` or `acquireToken` APIs normally, only setting the authority to the particular tenant you are requesting tokens for.
+
+> Note: For `login`/`acquireToken` interactive APIs, the tenant context is set from the authority's `tenantId`.
+
+```javascript
+const msalConfig = {
+    auth: {
+        clientId: "ENTER_CLIENT_ID",
+        authority: "https://login.microsoftonline.com/HOME_TENANT", // This is the authority that MSAL will default to for requests that don't specify their own authority.
+    },
+};
+
+const myMSALObj = new PublicClientApplication(msalConfig);
+
+// handleRedirectPromise has to be called to resume redirect requests
+myMSALObj
+    .handleRedirectPromise()
+    .then(handleResponse)
+    .catch((err) => {
+        console.error(err);
+    });
+
+// Authority isn't specified because it is already the default set in the MSAL Config object
+const homeTenantRequest = {
+    scopes: ["HOME_TENANT_SCOPE"],
+};
+
+// Guest tenant authority overrides the MSAL config authority, meaning the user can authenticate with the tenant they are a guest in and acquire tokens for it
+const guestTenantRequest = {
+    scopes: ["GUEST_TENANT_SCOPE"],
+    authority: "https://login.microsoftonline.com/GUEST_TENANT",
+};
+
+const homeTenantAuthResponse = await myMSALObj.loginPopup(homeTenantRequest);
+// Get the home tenant account from the AuthenticationResult
+const homeTenantAccount = homeTenantAuthResponse.account;
+
+const guestTenantAuthResponse = await myMSALObj.loginPopup(homeTenantRequest);
+// Get the guest tenant account from the AuthenticationResult
+const guestTenantAccount = guestTenantAuthResponse.account;
+```
+
+Assuming the user logs into both tenants with the same account, once both login requests are successfully completed and the user has authenticated with each tenant, the cache will contain:
+
+-   2 ID tokens (one per tenant)
+-   2 Access tokens (one per tenant)
+-   1 Refresh token (one per account, shared across tenants)
+-   1 AccountEntity containing both tenant profiles. The `homeTenantAccount` and `guestTenantAccount` objects shown above are both created from the same cached account entity augmented by the claims in their respective ID tokens.
```

Comment:
Cache details are technically implementation details and we don't actually guarantee this (especially in the case of RT when we start talking about any sort of brokering). Perhaps better to rework this to simply state that ID and Access tokens are tenant specific but RTs are not and skip cache discussion.